app/src/main/java/com/github/se/eduverse/viewmodel/FolderViewModel.kt

```diff
@@ -51,6 +62,7 @@ open class FolderViewModel(val repository: FolderRepository, val auth: FirebaseA
    */
   fun selectFolder(folder: Folder?) {
```

Comment:
Adding error handling here might be beneficial to ensure folder is not null before assigning it to `savedStateHandle`. This would provide additional robustness.